File: cci16.06-smallest-difference-lcci/python/Solution.py
class Solution:
    def smallestDifference(self, a: List[int], b: List[int]) -> int:
        a.sort()
        b.sort()
        n = len(a)
        m = len(b)
        minRet = float("inf")
        i = 0
        j = 0
        while i < n and j < m:
            if a[i] >= b[j]:
                minRet = min(minRet, a[i] - b[j])
                j += 1
            else:
                minRet = min(minRet, b[j] - a[i])
                i += 1
        return minRet

File: cci16.06-smallest-difference-lcci/javascript/Solution.js
/**
 * @param {number[]} a
 * @param {number[]} b
 * @return {number}
 */
var smallestDifference = function(a, b) {
    let sortFn = (i, j) => i - j
    a.sort(sortFn)
    b.sort(sortFn)
    let n = a.length
    let m = b.length
    let minRet = Number.MAX_SAFE_INTEGER
    let i = 0
    let j = 0
    while (i < n && j < m) {
        if (a[i] >= b[j]) {
            minRet = Math.min(minRet, a[i] - b[j])
            j++
        } else {
            minRet = Math.min(minRet, b[j] - a[i])
            i++
        }
    }
    return minRet
};